src/components/appSwitch/AppSwitch.js

```diff
@@ -241,6 +239,16 @@ const AppSwitch = (props: LoadingProps) => {
     init()
     navigateToUrlRef.current()
 
+    //handle the case user comes back from "connecting accounts"
+    AsyncStorage.getItem(GD_CONNECTPROVIDER).then(value => {
```

Comment:
this could be moved to the separate hook

```javascript
const useConnectedProviderDeepLink = (navigation, onMount = false) => {
  const { navigate } = navigation
  const { usesProvider, setProviderStateHash } = useConnectedProvider()

  const proceedWithDeepLink = useCallback(async () => {
    const isUsing = await usesProvider()

    if (!isUsing) {
      return
    }

    if (isMobileNative === false) {
      setProviderStateHash(DeepLinking.hash)
    }
    
    navigate('ConnectedAccounts')
  }, [usesProvider, setProviderStateHash, navigate])

  useEffect(() => {
    if (true !== onMount) {
      return
    }

    proceedWithDeepLink()
  }, [onMount, proceedWithDeepLink])

  return proceedWithDeepLink
}
```

you could not pass true to the second parameter and just call the function returned here